tester/runner.go

```diff
@@ -236,6 +249,19 @@ func (r *Runner) runTest(ctx context.Context, mod *ast.Module, rule *ast.Rule) (
 		}
 	} else if len(rs) == 0 {
 		tr.Fail = true
+		if bufFailureLineTracer != nil {
+			events := *bufFailureLineTracer
```

Comment:
I think it would be worth factoring this block into a separate function (e.g., `getFailedAtFromTrace(events)`).